lstchain/tools/lstchain_create_time_calibration_file.py

```diff
@@ -0,0 +1,97 @@
+"""
+Create drs4 time correction coefficients.
+"""
+import glob
+
+from ctapipe.core import Provenance, Tool, traits
+from ctapipe.io import EventSource
+from lstchain.calib.camera.time_correction_calculate import TimeCorrectionCalculate
+from tqdm.autonotebook import tqdm
+
+
+class TimeCalibrationHDF5Writer(Tool):
+    name = "TimeCalibrationHDF5Writer"
+    description = "Generate a HDF5 file with time calibration coefficients"
+
+    input = traits.Path(
+        help="Path to the fits.fz events file or directory to glob"
+    ).tag(config=True)
+
+    glob = traits.Unicode(
+        help="Filename pattern to glob files in the directory",
+        default_value="*"
+    ).tag(config=True)
+
+    progress_bar = traits.Bool(
+        help="show progress bar during processing",
+        default_value=True,
+    ).tag(config=True)
+
+    aliases = {
+        ("i", "input"): "TimeCalibrationHDF5Writer.input",
+        ("o", "output"): "TimeCorrectionCalculate.calib_file_path",
+        "glob": "TimeCalibrationHDF5Writer.glob",
+        "max-events": "EventSource.max_events",
+        "pedestal": "LSTR0Corrections.drs4_pedestal_path",
+        "dragon-reference-time": "EventTimeCalculator.dragon_reference_time",
+        "dragon-reference-counter": "EventTimeCalculator.dragon_reference_counter",
+    }
+
+    classes = [TimeCorrectionCalculate] + traits.classes_with_traits(EventSource)
```

Comment:
Does this script work with other files than LST observed raw data? If not, you could just explicitly use the `LSTEventSource` instead of `traits.classes_with_traits(EventSource)` to avoid showing options for event sources that are not supported by this tool